src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -81,6 +82,23 @@ public boolean hasPerson(Person person) {
         return persons.contains(person);
     }
 
+    /**
+     * Returns true if a {@code person} in the address book use this phone number.
+     */
+    public boolean hasPhone(Person person, Phone phone) {
+        requireNonNull(phone);
+        for (Person p : persons) {
```

Comment:
I feel that this check should be done in editPerson rather than here (reference to other fields like name). Instead, hasPhoneNumber should just take in Phone phone